test/e2e/service.go

```diff
@@ -232,28 +231,131 @@ var _ = Describe("Services", func() {
 		}()
 
 		name1 := "test1"
-		addEndpointPodOrFail(c, ns, name1, labels)
+		addEndpointPodOrFail(c, ns, name1, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name1)
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}})
 
 		name2 := "test2"
-		addEndpointPodOrFail(c, ns, name2, labels)
+		addEndpointPodOrFail(c, ns, name2, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name2)
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}, name2: {80}})
 
 		err = c.Pods(ns).Delete(name1, nil)
 		Expect(err).NotTo(HaveOccurred())
 		names = []string{name2}
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name2: {80}})
 
 		err = c.Pods(ns).Delete(name2, nil)
 		Expect(err).NotTo(HaveOccurred())
 		names = []string{}
 
-		validateEndpointsOrFail(c, ns, serviceName, expectedPort, names)
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{})
+
+		// We deferred Gingko pieces that may Fail, we aren't done.
+		defer func() {
+			close(done)
+		}()
+	}, 240.0)
+
+	It("should serve multiport endpoints from pods", func(done Done) {
+		// repacking functionality is intentionally not tested here - it's better to test it in an integration test.
+		serviceName := "multi-endpoint-test"
+		ns := api.NamespaceDefault
```

Comment:
Instead of default namespace, we have started doing
- use `createTestingNS` in the "BeforeEach", and then 
- delete the ns in "AfterEach".  
As an example, see something like `networking.go`